pkg/comp-functions/functions/spksmariadb/script/recreate.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+set -eox pipefail
+
+name="$STS_NAME"
+namespace="$STS_NAMESPACE"
+size="$STS_SIZE"
+release="$RELEASE_NAME"
+
+echo "Checking if the PVC sizes match"
+# Check if delete is necessary
+found=$(kubectl -n "$namespace" get sts "$name" -o json --ignore-not-found)
+
+foundsize=$(echo -En "$found" | jq -r '.spec.volumeClaimTemplates[] | select(.metadata.name=="data") | .spec.resources.requests.storage')
+
+if [[ $foundsize != "$size" ]]; then
+  echo "PVC sizes don't match, deleting sts"
```

Comment:
I would add the pause annotation at this point in the logic. This way it's contained within itself and less likely to run into issues.

Reply:
As mentioned above, this doesn't really work.